pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -88,7 +91,18 @@ func (d *PmemCSIDriver) Reconcile(r *ReconcileDeployment) (bool, error) {
 		// Deployment successfull, so no more reconcile needed for this deployment
 		return false, nil
 	case api.DeploymentPhaseRunning:
-		requeue, err := d.reconcileDeploymentChanges(r, oldDeployment, changes)
+		if !foundInCache {
```

Comment:
I made changes to handle this case. Now we handle all phases similar and pass through the same reconcile logic so that we "refresh" the sub-objects regardless of its phase.